src/lib/stitching/kaws/stitching.ts

```diff
@@ -117,9 +117,14 @@ export const kawsStitchingEnvironment = (
         resolve: (parent, _args, context, info) => {
           const query = parent.query
           const hasKeyword = Boolean(parent.query.keyword)
-          const contextWithCache = { ...context }
-          const filterArtworksLoader = context.filterArtworksLoaderWithCache
-          contextWithCache.filterArtworksLoader = filterArtworksLoader
+          const contextWithCache = {
+            ...context,
+            filterArtworksLoader: loaderParams =>
+              context.filterArtworksLoader(
+                { ...loaderParams },
+                { requestThrottleMs: 1000 * 60 * 60 } // 1 hour
```

Comment:
This isn't fully going to work. This breaks caching when the user is authenticated.

Authenticated requests resolve to a `filterArtworksLoader` that's authenticated and _skips caching_. So `context.filterArtworksLoader` here refers to that one. I think the clearest thing is to just introduce a new loader, that's unauthenticated and skips caching, and just use that explicitly here. AKA- the original approach.

Reply:
So even for authenticated people we don’t want personalised data to show up in these rails?

If so, we should just add `authenticatedLoaders` and `unauhtenticatedLoaders` to the context, aside from merging them as we already do, and then this code can refer to `context.unauhtenticatedLoaders.filterArtworksLoader`.

Reply:
Yea, makes sense. The further optimization we can make by using the unauthenticated loaders for requests that we know (via introspecting the filter params) aren't personalized, can similarly do `context.unauthenticatedLoaders..`.

Is there a downside to adding things in the context? I imagine if it's an in-memory map that unless you put a ton of data in there, then not really. Besides just being deliberate about what we add there, and trying to be tidy.

Reply:
Took a stab at this here: https://github.com/artsy/metaphysics/pull/1592

Reply:
@mzikherman I was actually thinking recently about measuring the amount of memory we use for each request that comes in and, if in any way significant, making the loader properties on the map lazy getters. Right now I don’t worry too much, but it’s not based on evidence.